gslib/commands/rsync.py

```diff
@@ -731,10 +731,14 @@ def _FieldedListingIterator(cls, gsutil_api, base_url_str, desc):
           'metadata/%s' % GID_ATTR,
           'metadata/%s' % UID_ATTR,
       ])
+    expanded_exclude_pattern = re.compile('{}/{}'.format(
+        base_url_str.rstrip('/\\'), cls.exclude_pattern.pattern
```

Comment:
Can we also add a test to ensure that rstrip was called? You can also modify the existing test to have a slash in the end of the base dir.

Reply:
Just added an integration test to rsync and included a slash in the base directory.

Reply:
Excellent! LGTM!